ts-node-plugin/index.js

```diff
@@ -0,0 +1,38 @@
+const {transform} = require("../dist");
+
+const ModuleKindCommonJS = 1;
+
+/**
+ * ts-node transpiler plugin
+ */
+function create(createOptions) {
```

Comment:
@cspotcode Here's what I have so far. Note that plenty of tsconfig options aren't supported by Sucrase (e.g. target ES level), so my plan is to just silently go with Sucrase defaults.

In terms of high-level plugin interface feedback, the two-level `create`/`transpile` approach doesn't seem particularly important for my use case, but I can certainly imagine it being useful for other cases. Currently I'm just doing all work in `transpile`, though I suppose I could front-load some of it.

Reply:
`nodeModuleEmitKind` is marked `@internal` in ts-node's API, so I should update to make this a public part of the API.  Will serve as documentation and as a reminder that we cannot make breaking changes.  I have called this out in https://github.com/TypeStrong/ts-node/issues/1851